src/App.tsx

```diff
@@ -1,12 +1,51 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { get5First, getAll, getRedGoods } from './api/goods';
+import { GoodsList } from './components/GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+
+  const getGoods = useCallback((get: () => Promise<Good[]>) => {
+    get()
+      .then(goodsFromServer => {
+        setGoods(goodsFromServer);
+      });
```

Comment:
async await gives you a simpler way to work with asynchronous promise-based code